### DISCUSSION TITLE
Resize TreeDataGrid Columns to auto when font size changes

### DISCUSSION THREAD

**Discussion by dariostrm** [AvaloniaUI/Avalonia#16614] [Q&A]
Hello, I have a TreeDataGrid in my app, but I want the user to be able to choose a font size so I have a slider in the same view that should control the font size.
I have the following styles
```c#
<TreeDataGrid.Styles>
                <Style Selector="TreeDataGridColumnHeader">
                    <Setter Property="FontSize" Value="{Binding #SliderFontSize.Value,
                    Converter={StaticResource MultiplicationConverter}, ConverterParameter=0.7}" />
                    <Setter Property="VerticalAlignment" Value="Center" />
                </Style>
                <Style Selector="TreeDataGridTextCell">
                    <Setter Property="FontSize" Value="{Binding #SliderFontSize.Value,
                    Converter={StaticResource MultiplicationConverter}, ConverterParameter=0.7}" />
                    <Setter Property="VerticalAlignment" Value="Center" />
                </Style>
                <Style Selector="TreeDataGridTemplateCell">
                    <Setter Property="FontSize" Value="{Binding #SliderFontSize.Value,
                    Converter={StaticResource MultiplicationConverter}, ConverterParameter=0.7}" />
                    <Setter Property="VerticalAlignment" Value="Center" />
                </Style>
                <Style Selector="TreeDataGridTemplateCell">
                    <Setter Property="FontSize" Value="{Binding #SliderFontSize.Value,
                    Converter={StaticResource MultiplicationConverter}, ConverterParameter=0.7}" />
                    <Setter Property="VerticalAlignment" Value="Center" />
                </Style>
            </TreeDataGrid.Styles>
```

And these do resize all the text in the tree data grid, but the columns stay very wide when I decrease the font size, and double-clicking a column still makes it as wide as it was before resizing the font size. Is there a way to have the columns resize when I change the font size to just fit the content?

**Comment by Nick-Crane** [comment]
Much easier to use a LayoutTransformControl.

        <LayoutTransformControl>
            <LayoutTransformControl.LayoutTransform>
                
                <ScaleTransform 
                     ScaleX="{Binding ZoomFactor}" 
                     ScaleY="{Binding ZoomFactor}" />
            </LayoutTransformControl.LayoutTransform>
            <TreeDataGrid ... />
        </LayoutTransformControl>
